src/QSymbolicsBase/basic_ops_homogeneous.jl

```diff
@@ -82,7 +87,73 @@ Base.show(io::IO, x::SAddOperator) = print(io, "("*join(map(string, arguments(x)
 const SAddBra = SAdd{AbstractBra}
 Base.show(io::IO, x::SAddBra) = print(io, "("*join(map(string, arguments(x)),"+")::String*")") # type assert to help inference
 
-"""Tensor product of quantum objects (kets, operators, or bras)."""
+# defines commutativity for addition of quantum objects
+const CommQObj = Union{SAddBra, SAddKet, SAddOperator} 
+function _in(x::SymQObj, sadd::CommQObj)
+    for i in arguments(sadd)
+        if isequal(x, i)
+            return true
+        end
+    end
+    false
+end
+function Base.isequal(x::CommQObj, y::CommQObj)
+    if typeof(x)==typeof(y)
+        if isexpr(x)
+            if operation(x)==operation(y)
+                ax,ay = arguments(x),arguments(y)
+                (length(ax) == length(ay)) && all(x -> _in(x, y), ax)
```

Comment:
`Set` doesn't appear to work here, as equality of sets relies on `==` rather than `isequal` function. Here's an example:
```
julia> A = SOperator(:A, SpinBasis(1//2)); B = SOperator(:B, SpinBasis(1//2));

julia> Set([A+B]) == Set([B+A])
false

julia> isequal(A+B, B+A)
true

julia> A+B == B+A
false
```

Reply:
I defined set containment of quantum objects with `_in(x::SymQObj, y::SymQObj)`. Performance-wise, it's probably not better, but unless we want to redefine `==` for quantum objects, it might be better to stick with this option.